python/lsst/ctrl/oods/butlerBroker.py

```diff
@@ -0,0 +1,135 @@
+# This file is part of ctrl_oods
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import asyncio
+import os
+import os.path
+from pathlib import PurePath
+from abc import ABC, abstractmethod
+
+
+class ButlerBroker(ABC):
+    """Interface class for processing files for a butler.
+    """
+
+    SUCCESS = 0
+    FAILURE = 1
+
+    @abstractmethod
+    def ingest(self, file_list):
+        """Placeholder to ingest a file
+
+        Parameters
+        ----------
+        file_list: `list`
+            list of files to ingest
+        """
+        raise NotImplementedError()
+
+    @abstractmethod
+    def getName(self):
+        """Get the name of this ingester
+
+        Returns
+        -------
+        ret: `str`
+            the name of this ingester
+        """
+        raise NotImplementedError()
+
+    async def cleaner_task(self):
+        """Run task that require periodical attention
+        """
+        await asyncio.sleep(60)
+
+    def clean(self):
+        """Perform a cleaning pass for this ingester; override if necessary
+        """
+        pass
```

Comment:
If it is up to the derived class to provide the implementation of this method, why it is not an abstract method?

Reply:
In this particular case, not all subclasses will implement the clean() method.  for Gen2 butler, the cleaning is handled externally (along with all other directories external to the OODS that need to be cleaned up). so in this case it's a pass.